vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
+        		    }else{
+        		        unpublishedSpaceElement.css('left', link.positionX + posX + 25);
+            		    unpublishedSpaceElement.css('top', link.positionY + posY - 13);
+        		    }
+        		    unpublishedSpaceElement.css('transform', 'rotate('+link.rotation+'deg)');
+        		    unpublishedSpaceElement.css('font-size', "12px");
+        		    if(isModal){
+        		    	$("#space").append(unpublishedSpaceElement);
+        		    }else{
+        		        $("#main-space").append(unpublishedSpaceElement);
+        		    }
+        		}
+                if(isExternalLink){
+                    setLabelPosition(link,link.externalLink.id,isModal);
+                }else{
+                    setLabelPosition(link,link.link.id,isModal);
+                }
+            }
+            function setLabelPosition(link,id, isModal){
+                if(isModal){
+                    $(".label-"+id).css({
+            			'left': link.positionX + posXModal - 10,
+            			'top': link.positionY + posYModal + 35,
+            			'color': 'red'
+            		});	
+                }
+                else{
+                    $(".label-"+id).css({
+                		'left': link.positionX + posX - 10,
+                		'top': link.positionY + posY + 35,
+                		'color': 'red'
+                	});
+                }
+            }
+            function setInputElements(link,isModal,isExternalLink){
+                var inputLabel;
+                var inputType;
+                var inputRotation;
+                if(isExternalLink){
+                    inputLabel = document.createElement("input");
+                	inputLabel.setAttribute("type", "hidden");
+                	inputLabel.setAttribute("id", "linkLabel-"+link.externalLink.id);
+                	inputLabel.setAttribute("value", link.externalLink.name);
+                	inputType = document.createElement("input");
+                	inputType.setAttribute("type", "hidden");
+                	inputType.setAttribute("id", "linkType-"+link.externalLink.id);
+                	inputType.setAttribute("value", link.type);
+                }else{
+                    inputRotation = document.createElement("input");
+            		inputRotation.setAttribute("type", "hidden");
+            		inputRotation.setAttribute("id", "linkRotation-"+link.link.id);
+            		inputRotation.setAttribute("value", link.rotation);
+            		inputLabel = document.createElement("input");
+            		inputLabel.setAttribute("type", "hidden");
+            		inputLabel.setAttribute("id", "linkLabel-"+link.link.id);
+            		inputLabel.setAttribute("value", link.link.name);
+            		inputType = document.createElement("input");
+            		inputType.setAttribute("type", "hidden");
+            		inputType.setAttribute("id", "linkType-"+link.link.id);
+            		inputType.setAttribute("value", link.type);

```

Comment:
what are these inputs for? If they are just for keeping link information that should be done via data attributes.